plugins/rebalance/rest.go

```diff
@@ -0,0 +1,364 @@
+package rebalance
+
+import (
+	"fmt"
+	"net/http"
+
+	"github.com/gluster/glusterd2/glusterd2/gdctx"
+	restutils "github.com/gluster/glusterd2/glusterd2/servers/rest/utils"
+	"github.com/gluster/glusterd2/glusterd2/transaction"
+	"github.com/gluster/glusterd2/glusterd2/volume"
+	"github.com/gluster/glusterd2/pkg/api"
+	"github.com/gorilla/mux"
+	"github.com/pborman/uuid"
+	log "github.com/sirupsen/logrus"
+)
+
+// RebalanceStatus represents Rebalance Status
+type RebalanceStatus uint64
+
+const (
+	// GfDefragStatusNotStarted should be set only for a volume in which rebalance process is not started
+	GfDefragStatusNotStarted RebalanceStatus = iota
+	// GfDefragStatusStarted should be set only for a volume that has been just started rebalance process
+	GfDefragStatusStarted
+	// GfDefragStatusStopped should be set only for a volume that has been just stopped rebalance process
+	GfDefragStatusStopped
+	// GfDefragStatusComplete should be set only for a volume that the rebalance process is completed
+	GfDefragStatusComplete
+	// GfDefragStatusFailed should be set only for a volume that are failed to run rebalance process
+	GfDefragStatusFailed
+	// GfDefragStatusLayoutFixStarted should be set only for a volume that has been just started rebalance fix-layout
+	GfDefragStatusLayoutFixStarted
+	// GfDefragStatusLayoutFixStopped should be set only for a volume that has been just stopped rebalance fix-layout
+	GfDefragStatusLayoutFixStopped
+	// GfDefragStatusLayoutFixComplete should be set only for a volume that the rebalance fix-layout is completed
+	GfDefragStatusLayoutFixComplete
+	// GfDefragStatusLayoutFixFailed should be set only for a volume that are failed to run rebalance fix-layout
+	GfDefragStatusLayoutFixFailed
+)
+
+// RebalanceInfo represents Rebalance details
+type RebalanceInfo struct {
+	Volname           string
+	Status            RebalanceStatus
+	RebalanceID       uuid.UUID
+	RebalanceFiles    uint64
+	RebalanceData     uint64
+	LookedupFiles     uint64
+	RebalanceFailures uint64
+	ElapsedTime       uint64
+	SkippedFiles      uint64
+	TimeLeft          uint64
+	CommitHash        uint64
+}
+
+func createRebalanceInfo(rebalanceinfo *RebalanceInfo) (*RebalanceInfo, error) {
+	v := new(RebalanceInfo)
+	v.Volname = rebalanceinfo.Volname
+	v.RebalanceID = uuid.NewRandom()
+	v.Status = rebalanceinfo.Status
+	v.CommitHash = rebalanceinfo.CommitHash
+	return v, nil
+}
+
+func rebalanceStartInfoResp(v *RebalanceInfo) *api.RebalanceInfo {
+
+	return &api.RebalanceInfo{
+		Volname:     v.Volname,
+		Status:      api.RebalanceStatus(v.Status),
+		RebalanceID: v.RebalanceID,
+	}
+}
+
+func registerRebalanceStartStepFuncs() {
+	transaction.RegisterStepFunc(StartRebalance, "rebal-start.Commit")
```

Comment:
You can add all these register calls inside `RegisterStepFuncs()` itself, instead of making as separate funcs